_pages/clients/curl.md

```diff
@@ -15,78 +15,171 @@ redirect_from:
 
-## Working with Data via cURL
+You might have noticed that in this example we included a version number when instantiating the algorithm. Pinning your code to a specific version of the algorithm can be especially important in a production environment where the underlying implementation might change from version to version.
 
-#### Specifying an On-Premises or Private Cloud Endpoint
+### Request Options
 
-This feature is available to [Algorithmia Enterprise](/enterprise) users only.
-{: .notice-enterprise}
+The API exposes options that can configure algorithm requests. This includes support for changing the timeout or indicating that the API should include stdout in the response. From cURL, we can provide these options as URL parameters. In the following example, we set the timeout to 60 seconds and disable `stdout` in the response:
 
-If you are running [Algorithmia Enterprise](/enterprise), replace `https://api.algorithmia.com/v1/data/.my` in the codesamples below with your own API endpoint URL.
+{% highlight bash %}
+curl 'https://api.algorithmia.com/v1/algo/demo/Hello?timeout=60&stdout=false' \
+    -X POST \
+    -H 'Content-Type: application/json' \
+    -H 'Authorization: Simple YOUR_API_KEY' \
+    -d '"HAL 9001"'
+{% endhighlight %}
 
+You can find more details in [API Docs](/developers/api/?shell) > [Invoke an Algorithm](/developers/api/?shell#invoke-an-algorithm).
 
-#### Create a directory
+### Error Handling
 
-You can use cURL to interact with the Data API from the command line.
-To create a directory, POST the new directory name to the Data API URI.
-You will get a result that returns the address of the new data collection.
+To be able to better develop across languages, Algorithmia has created a set of standardized errors that can be returned by either the platform or by the algorithm being run. If an error occurs while invoking the API, the HTTP response will include an error field with more information:
 
 {% highlight bash %}
-curl -X POST -d '{"name": "newCollection"}' -H 'Content-Type: application/json' -H 'Authorization: Simple YOUR_API_KEY' https://api.algorithmia.com/v1/data/.my
-
--> {"result": "data://.my/newCollection"}
+curl https://api.algorithmia.com/v1/algo/util/whoopsWrongAlgo \
+    -X POST \
+    -H 'Content-Type: application/json' \
+    -H 'Authorization: Simple YOUR_API_KEY' \
+    -d '"Hello, world"'
 {% endhighlight %}
 
-#### Upload File
+Response: `{"error":{"message":"algorithm algo://util/whoopsWrongAlgo/ not found"}}`
+
+You can read more about [Error Handling](/developers/algorithm-development/algorithm-errors) in the [Algorithm Development](/developers/algorithm-development) section of the dev center.
+
+### Limits
+
+Your account can make up to {{site.data.stats.platform.max_num_algo_requests}} Algorithmia requests at the same time (this limit <a onclick="Intercom('show')">can be raised</a> if needed).
```

Comment:
`Intercom` is undefined in the global scope in prod. Let's change the link slightly:
```suggestion
Your account can make up to {{site.data.stats.platform.max_num_algo_requests}} Algorithmia requests at the same time (this limit <a href="mailto:support@algorithmia.com?subject=Request to raise max algorithm request limit">can be raised</a> if needed).
```